Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,653 @@
-						}
+						return convertIfcSurfaceOfLinearExtrusion(
+							surface.as<typename IfcEntityTypesT::IfcSurfaceOfLinearExtrusion>(),
+							pos,
+							polylineData);
+					}
 
-						// (3/4) IfcCurveBoundedSurface SUBTYPE of IfcBoundedSurface.
-						std::shared_ptr<typename IfcEntityTypesT::IfcCurveBoundedSurface> curve_bounded_surface =
-							std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcCurveBoundedSurface>(bounded_surface);
+					// (2/2) IfcSurfaceOfRevolution SUBTYPE of IfcSweptSurface
+					else if (surface.isOfType<typename IfcEntityTypesT::IfcSurfaceOfRevolution>()) {
 
-						if(curve_bounded_surface) {
-							// Get basis surface, boundaries and implicit outer.
-							/*std::shared_ptr<typename IfcEntityTypesT::IfcSurface> basis_surface =
-								std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcSurfacePlane>(curve_bounded_surface);*/
-							std::shared_ptr<typename IfcEntityTypesT::IfcSurface> basis_surface = curve_bounded_surface->BasisSurface.lock();
-							/*std::shared_ptr<typename IfcEntityTypesT::IfcBoundaryCurve> boundaries =
-								std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcBoundaryCurve>(curve_bounded_surface);*/
-								// std::shared_ptr<typename IfcEntityTypesT::IfcBoundaryCurve> boundaries = curve_bounded_surface->Boundaries;
+						return convertIfcSurfaceOfRevolution(
+							surface.as<typename IfcEntityTypesT::IfcSurfaceOfRevolution>(),
+							pos,
+							polylineData);
+					}
+				}
 
-							std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcBoundaryCurve>> vec_boundaries;
+				//--------------------------------------------------------------------------------------------
+				// Conversion functions
+				//--------------------------------------------------------------------------------------------
 
-							vec_boundaries.resize(curve_bounded_surface->Boundaries.size());
-							std::transform(
-								curve_bounded_surface->Boundaries.begin(), curve_bounded_surface->Boundaries.end(), vec_boundaries.begin(), [](auto& it) { return it.lock(); });
-							// std::shared_ptr<typename IfcEntityTypesT::IfcCartesianPoint> control_point_list = -bspline_surface->ControlPointsList;
-							/*std::shared_ptr<typename IfcEntityTypesT::IfcBoolean> implicit_outer =
-								std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcBoolean>(curve_bounded_surface);*/
-							bool implicit_outer = curve_bounded_surface->ImplicitOuter;
+					/*! \brief  Converts \c IfcBsplineSurface to ...
+						\param	surface	\c IfcBSplineSurface entity to be interpreted.
+						\param	pos
+						\param	polylineData
+						\note	The \c IfcBSplineSurface is a subtype of \c IfcBoundedSurface.
+					*/
 
-							// TODO: implement		// Interpret values and calculate.
+				void convertIfcBSplineSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcBsplineSurface>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...) {
+

```

Comment:
Same. @christophKaiser This might also be interesting for you!

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/1ee08611e218ac8b03e6fabd0e214cad06a096b1